src/OutboundServer/api.yaml

```diff
@@ -581,6 +637,73 @@ components:
         transferRequestExtensions:
           $ref: '#/components/schemas/extensionList'
 
+    requestToPayInboundRequest:
+      type: object
+      required:
+        - transactionRequestId
+        - from
+        - to
+        - amountType
+        - currency
+        - amount
+        - scenario
+        - initiator
+        - initiatorType
+      properties:
+        transactionRequestId:
+          type: string
+          description: Transaction ID from the DFSP backend, used to reconcile transactions between the Switch and DFSP backend systems.
+        from:
+          $ref: '#/components/schemas/transferParty'
+        to:
+          $ref: '#/components/schemas/transferParty'
+        amountType:
+          $ref: '#/components/schemas/amountType'
+        currency:
+          $ref: '#/components/schemas/currency'
+        amount:
+          $ref: '#/components/schemas/money'
+        scenario:
+          $ref: '#/components/schemas/scenario'
+        initiator:
+          $ref: '#/components/schemas/initiator'
+        initiatorType:
+          $ref: '#/components/schemas/initiatorType'
+
+    requestToPayResponse:
```

Comment:
section 6.4.3.1 of API spec document suggests the response for a transactionRequest will contain a "transactionRequestState" object. I think you probably want to model and expose that to the DFSP backend in some way, rather than simply reflecting the request back to them.

Reply:
yeah..good point...the payee should definitely know the status of this request so that he can proceed to next steps if it was accepted.

Reply:
@bushjames ... I just checked the response being returned to Backend. The final state is indeed returned back.

{
    "transactionRequestId": "cd04d5fc-8328-4ca6-9761-fce75b1a3e2d",
    "from": {
        "idType": "BUSINESS",
        "idValue": "Starbucks",
        "displayName": "Starbucks Coffee"
    },
    "to": {
        "idType": "MSISDN",
        "idValue": "17039811907",
        "fspId": "payerfsp",
        "firstName": "GoldenFirst",
        "middleName": "GoldenMiddle",
        "lastName": "GoldenLast",
        "dateOfBirth": "2010-10-10"
    },
    "amountType": "SEND",
    "currency": "USD",
    "amount": "10",
    "scenario": "PAYMENT",
    "initiator": "PAYEE",
    "initiatorType": "BUSINESS",
    "note": "test payment",
    "currentState": "COMPLETED",
    "transactionRequestResponse": {
        "transactionRequestState": "RECEIVED"
    }
}

Reply:
@bushjames are you suggesting "transactionRequestState": "RECEIVED" to be at the root level instead of within transactionRequestResponse?

Reply:
@bushjames added a new field to represent the final state, requestToPayState